mmengine/fileio/backends/oss_backend.py

```diff
@@ -0,0 +1,809 @@
+        f'does not equals src file size {total_size}, please recopy.'
+        return str(dst)
+
+    def list_dir_or_file(self,
+                         dir_path: Union[str, Path],
+                         list_dir: bool = True,
+                         list_file: bool = True,
+                         suffix: Optional[Union[str, Tuple[str]]] = None,
+                         recursive: bool = False) -> Iterator[str]:
+        """Scan a directory to find the interested directories or files in
+        arbitrary order.
+
+        Note:
+            OSS has no concept of directories but it simulates the directory
+            hierarchy in the filesystem through public prefixes. In addition,
+            if the returned path ends with '/', it means the path is a public
+            prefix which is a logical directory.
+
+        Note:
+            :meth:`list_dir_or_file` returns the path relative to ``dir_path``.
+            In addition, the returned path of directory will contains the
+            suffix '/' which is inconsistent with other backends.
+
+        Args:
+            dir_path (str | Path): Path of the directory.
+            list_dir (bool): List the directories. Defaults to True.
+            list_file (bool): List the path of files. Defaults to True.
+            suffix (str or tuple[str], optional):  File suffix
+                that we are interested in, such as .txt, .jpg etc.
+                Defaults to None.
+            recursive (bool): If set to True, recursively scan the
+                directory. Defaults to False.
+
+        Yields:
+            Iterable[str]: A relative path to ``dir_path``.
+
+        Examples:
+            >>> backend = OSSBackend()
+            >>> dir_path = 'oss://endpoint/bucket/file/'
+            >>> # list those files and directories in current directory
+            >>> for file_path in backend.list_dir_or_file(dir_path):
+            ...     print(file_path)
+            >>> # only list files
+            >>> for file_path in backend.list_dir_or_file(dir_path,
+                list_dir=False):
+            ...     print(file_path)
+            >>> # only list directories
+            >>> for file_path in backend.list_dir_or_file(dir_path,
+                list_file=False):
+            ...     print(file_path)
+            >>> # only list files ending with specified suffixes
+            >>> for file_path in backend.list_dir_or_file(dir_path,
+                suffix='.txt'):
+            ...     print(file_path)
+        """
+
+        endpoint, bucket_name, obj_name = self._parse_path(dir_path)
+        bucket = self._bucket_instance(endpoint, bucket_name)
+        if list_dir and suffix is not None:
+            raise TypeError(
+                '`list_dir` should be False when `suffix` is not None')
+
+        root_path = obj_name
+        if root_path and not root_path.endswith('/'):
+            raise TypeError('`dir_path` must endswith "/" ')
```

Comment:
After implementing the `is_dir` method, there is no need to require `root_path` ending with `/`.